presto-main/src/main/java/com/facebook/presto/connector/ConnectorManager.java

```diff
@@ -195,10 +199,13 @@ private synchronized void addCatalogConnector(String catalogName, ConnectorId co
                 new InformationSchemaConnector(catalogName, nodeManager, metadataManager, accessControlManager));
 
         ConnectorId systemId = createSystemTablesConnectorId(connectorId);
+        SystemTablesProvider staticProvider = new StaticSystemTablesProvider(connector.getSystemTables());
+        SystemTablesProvider metadataBasedProvider = new MetadataBasedSystemTablesProvider(metadataManager, catalogName);
+        SystemTablesProvider systemTablesProvider = new DelegatingSystemTablesProvider(staticProvider, metadataBasedProvider);
```

Comment:
I'd format this like
```java
SystemTablesProvider systemTablesProvider = new DelegatingSystemTablesProvider(
        new StaticSystemTablesProvider(connector.getSystemTables()),
        new MetadataBasedSystemTablesProvider(metadataManager, catalogName));
```
On first read, I wondered why we were creating three of them.

Reply:
Might be as one class. @kokosing i don't understand your vote, however. It was you who split `CachingStatsProvider` into 2 classes ;)

Reply:
Lets keep it as it is. Both ways are simple, so there is no big deal which way we go.